opm/parser/eclipse/EclipseState/Tables/tests/TableTests.cpp

```diff
@@ -355,3 +357,686 @@ BOOST_AUTO_TEST_CASE(PvtoTable_Tests) {
+VFPPROD \n\
+-- Table Depth  Rate   WFR   GFR      \n\
+-- ----- ----- ----- ----- -----      \n\
+      5  32.9  'LIQ' 'WCT' 'GOR'    / \n\
+-- Rate axis \n\
+1 2 /        \n\
+-- THP axis  \n\
+7 /          \n\
+-- WFR axis  \n\
+13 /         \n\
+-- GFR axis  \n\
+19 /         \n\
+-- ALQ axis  \n\
+29 /         \n\
+-- Table data with THP# WFR# GFR# ALQ# <values 1-num_rates> \n\
+-- Will fail, as rate axis requires two elements            \n\
+1 1 1 1 1.5 /    \n";
+
+        Opm::ParserPtr parser(new Opm::Parser);
+        Opm::DeckConstPtr deck(parser->parseString(missing_values));
+        Opm::DeckKeywordConstPtr vfpprodKeyword = deck->getKeyword("VFPPROD");
+        std::shared_ptr<Opm::UnitSystem> units(Opm::UnitSystem::newMETRIC());
+        BOOST_CHECK_EQUAL(deck->numKeywords("VFPPROD"), 1);
+
+        Opm::VFPProdTable vfpprodTable;
+
+
+        BOOST_CHECK_THROW(vfpprodTable.init(vfpprodKeyword, units), std::invalid_argument);
+    }
+
+
+
+    /**
+     * Missing value in table #2
+     */
+    {
+        const char *missing_values = "\
+VFPPROD \n\
+-- Table Depth  Rate   WFR   GFR      \n\
+-- ----- ----- ----- ----- -----      \n\
+      5  32.9  'LIQ' 'WCT' 'GOR'    / \n\
+-- Rate axis \n\
+1 /          \n\
+-- THP axis  \n\
+7 9 /        \n\
+-- WFR axis  \n\
+13 /         \n\
+-- GFR axis  \n\
+19 /         \n\
+-- ALQ axis  \n\
+29 /         \n\
+-- Table data with THP# WFR# GFR# ALQ# <values 1-num_rates> \n\
+-- Will fail, as two entries are required                   \n\
+1 1 1 1 1.5 /    \n";
+
+        Opm::ParserPtr parser(new Opm::Parser);
+        Opm::DeckConstPtr deck(parser->parseString(missing_values));
+        Opm::DeckKeywordConstPtr vfpprodKeyword = deck->getKeyword("VFPPROD");
+        std::shared_ptr<Opm::UnitSystem> units(Opm::UnitSystem::newMETRIC());
+        BOOST_CHECK_EQUAL(deck->numKeywords("VFPPROD"), 1);
+
+        Opm::VFPProdTable vfpprodTable;
+
+
+        BOOST_CHECK_THROW(vfpprodTable.init(vfpprodKeyword, units), std::invalid_argument);
+    }
+
+
+    /**
+     * Missing items in header
+     */
+    {
+        const char *missing_metadata = "\
+VFPPROD \n\
+-- Table Depth   \n\
+-- ----- -----   \n\
+      5  32.9  / \n\
+-- Rate axis \n\
+1 2 /        \n\
+-- THP axis  \n\
+7 /          \n\
+-- WFR axis  \n\
+13 /         \n\
+-- GFR axis  \n\
+19 /         \n\
+-- ALQ axis  \n\
+29 /         \n\
+-- Table data with THP# WFR# GFR# ALQ# <values 1-num_rates> \n\
+1 1 1 1 1.5 2.5 /    \n";
+
+        Opm::ParserPtr parser(new Opm::Parser);
+        Opm::DeckConstPtr deck(parser->parseString(missing_metadata));
+        Opm::DeckKeywordConstPtr vfpprodKeyword = deck->getKeyword("VFPPROD");
+        std::shared_ptr<Opm::UnitSystem> units(Opm::UnitSystem::newMETRIC());
+        BOOST_CHECK_EQUAL(deck->numKeywords("VFPPROD"), 1);
+
+        Opm::VFPProdTable vfpprodTable;
+
+
+        BOOST_CHECK_THROW(vfpprodTable.init(vfpprodKeyword, units), std::invalid_argument);
```

Comment:
This check fails for me: it throws
```
std::out_of_range: Index must be smaller than 0 but is 0
```
from something called from line 87 of VFPProdTable.cpp:
```
m_flo_type = getFloType(header->getItem<VFPPROD::RATE_TYPE>()->getString(0));
```
whereas the check specifies an std::invalid_argument.